packages/story-editor/src/components/library/panes/pageTemplates/templateSave.js

```diff
@@ -96,26 +98,67 @@ const SaveButton = styled.button`
 `;
 
 function TemplateSave({ setShowDefaultTemplates, updateList }) {
+  const {
+    capabilities: { hasUploadMediaAction },
+  } = useConfig();
   const {
     actions: { addPageTemplate },
   } = useAPI();
+  const {
+    actions: { uploadFile },
+  } = useUploader();
   const { showSnackbar } = useSnackbar();
 
   const { currentPage } = useStory(({ state: { currentPage } }) => ({
     currentPage,
   }));
 
+  const { fullbleedContainer } = useCanvas((state) => ({
+    fullbleedContainer: state.state.fullbleedContainer,
+  }));
+
   const isDisabled = useMemo(
     () => currentPage && isDefaultPage(currentPage),
     [currentPage]
   );
   const handleSaveTemplate = useCallback(
-    (e) => {
+    async (e) => {
       e.preventDefault();
+
       if (isDisabled) {
         return;
       }
-      addPageTemplate({ ...currentPage, id: uuidv4(), title: null })
+
+      let imageId;
+
+      if (hasUploadMediaAction) {
+        // TODO(#9574): Generate image using html-to-image here.
+
+        const htmlToImage = await import(
+          /* webpackChunkName: "chunk-html-to-image" */ 'html-to-image'
+        );
+
+        // TODO: We need something more reliable than fullbleedContainer. What if the canvas is small or zoomed in?
```

Comment:
🤔  could it make sense to treat getting the screenshot of a custom template like optimizing media? So you add a page as a custom template and then that shows the same preview node we're used to seeing in the custom page templates view (but maybe we only display the safe area), while this is rendered we show the loading/progress bar in the UI and use the same fallback code you outlined earlier in this PR to grab the screenshot. This would eliminate weird situations where the canvas is zoomed in or super small since the panel won't shrink below 360px, it would also unify the process regardless of backfill/newly added template.